static/app/views/dashboards/editAccessSelector.tsx

```diff
@@ -268,6 +274,19 @@ function EditAccessSelector({
       isOpen={isMenuOpen}
       onOpenChange={() => {
         setMenuOpen(!isMenuOpen);
+        if (!isMenuOpen) {
+          teamsToRender.sort((team1, team2) => {
+            const isTeam1Selected = selectedOptions.includes(team1.id);
+            const isTeam2Selected = selectedOptions.includes(team2.id);
+            if (isTeam1Selected && !isTeam2Selected) {
+              return -1;
+            }
+            if (!isTeam1Selected && isTeam2Selected) {
+              return 1;
+            }
+            return team1.name.localeCompare(team2.name);
+          });
+        }
```

Comment:
I'm noticing a lot of lag when opening and closing the menu and I _think_ it might be related to all of the sorting we're adding here. A viable alternative approach is to just define the selected teams in the `allDropdownOptions` options like here: https://github.com/getsentry/sentry/pull/81494/files#diff-39e9d98d74f1ddbcc9d1b7bcd122c6b157da5054e2c7d61d59eba14208a8f073R199

e.g. `options: [...selectedTeam, ...teamsToRender].map(makeTeamOption),`, but you'll have to filter `selectedTeam`s out of the `teamsToRender` or else we get duplicates. I tried this out locally and I think it performs better, would you be able to try this out?